src/code42cli/cmds/search/extraction.py

```diff
@@ -61,12 +79,18 @@ def handle_response(response):
                 events = _get_alert_details(sdk, events)
             except Exception as ex:
                 handlers.handle_error(ex)
-        handlers.TOTAL_EVENTS += len(events)
-        event = None
-        for event in events:
-            output_logger.info(event)
-        if event:
-            last_event_timestamp = extractor._get_timestamp_from_item(event)
+
+        total_events = len(events)
+        handlers.TOTAL_EVENTS += total_events
+
+        for events_per_page in _paginate(events, _EVENT_COUNT_PER_PAGE):
+            output = _process_events(
+                output_format, include_all, events_per_page, output_header
+            )
+            click.echo_via_pager(output)
+
+        if total_events:
+            last_event_timestamp = extractor._get_timestamp_from_item(events[-1])
```

Comment:
This could be dangerous. We may change the extractor internal method not realizing or remembering it is being used here.

Reply:
I agree, but this was already in place. My hope is that once we have proper support for getting more than 10K events via the api that we can just kill the extractor altogether.

Reply:
This already happens in the extractor.

Reply:
Any reason for that? @alanag13 @timabrmsn

Reply:
We just found this part is duplicated, those two lines are already being executed in "extractor" and things work just fine event without it. So we have removed it.

Reply:
It was duplicated here because when you hit ctrl-c to interrupt the process, it will finish processing the _current_ batch (the existing `handle_response()` call), then exit. But the extractor only saves the checkpoint _after_ `handle_response()` is called. So if you ctrl-c you'd get a bunch of processed events but the checkpoint would be off and the next run you'd get duplicates.

Reply:
Added back.